ioread.cpp

```diff
@@ -1070,9 +1070,22 @@ bool exec_readGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
    phiprof::start("readScalars");
 
    vlsv::ParallelReader file;
-   MPI_Info mpiInfo = MPI_INFO_NULL;
 
-   if (file.open(name,MPI_COMM_WORLD,MASTER_RANK,mpiInfo) == false) {
+   MPI_Info MPIinfo;
+   if (P::restartReadHints.size() == 0) {
+      MPIinfo = MPI_INFO_NULL;
+   } else {
+      MPI_Info_create(&MPIinfo);
+      
+      for (std::vector<std::pair<std::string,std::string>>::const_iterator it = P::restartReadHints.begin();
```

Comment:
This would be a prime example for a loop, that would be much more readable as a range-based loop, like
```
for (i : P::restartReadHints) {
   MPI_Info_set(MPIinfo, i.first.c_str(), i.second.c_str());
}
```

Reply:
It's copied from the other two such instances. And I hate those `auto`s, must be getting old'n'grumpy. But if you so wish, I shall change the three instances.

Reply:
Nah, it's fine.